### PR TITLE
tests/common: refactor to reduce duplicate code

### PR DESCRIPTION
**Pull request by jhscheer**
* move fn expected_result to common/util.rs
* move fn check_coreutil_version to common/util.rs
* move fn whoami to common/util.rs
* move macro unwrap_or_return to common/macros.rs
* add documentation
* add tests

### CODE REVIEW

File: tests/by-util/test_whoami.rs [hunk 1]
```diff
@@ -1,63 +1,47 @@
...
-        if is_ci() && result.stderr_str().contains(needle) {
-            println!("test skipped:");
-            return true;
-        } else {
-            result.success();
-        }
-    }
-    false
-}
+use crate::common::util::*;
```
Comment (sylvestre) [inline tests/by-util/test_whoami.rs:6]: could you please fix that warning? thanks

File: tests/common/util.rs [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+//  * file that was distributed with this source code.
+
+//spell-checker: ignore (linux) rlimit prlimit Rlim coreutil
 
 #![allow(dead_code)]
 
 use pretty_assertions::assert_eq;
 #[cfg(target_os = "linux")]
 use rlimit::{prlimit, rlim};
+use std::borrow::Cow;
```
Comment (sylvestre) [inline tests/common/util.rs:13]: this one too

File: tests/by-util/test_whoami.rs (hunk 1 above)
Comment (jhscheer) [reply tests/by-util/test_whoami.rs:7]: done

File: tests/common/util.rs (hunk 2 above)
Comment (jhscheer) [reply tests/common/util.rs:14]: done

**Pull request merged**